tests/datadriven/test_variants.py

```diff
@@ -84,19 +152,52 @@ def verifyVariantsEqual(self, gaVariants, pyvcfVariants):
                     self.assertEqual(str(alt1), str(alt2))
             else:
                 self.assertEqual(gaVariant.alternateBases, alt)
-            # TODO check INFO fields
 
-    def testSearchAllVariants(self):
-        allVariants = []
+            pyvcfCallMap = {}
+            for call in pyvcfVariant:
+                pyvcfCallMap[call.sample] = call
+            _verifyVariantCalls()
+
+    def _verifyGaVariantsSample(self, gaVariants, sampleIds):
+        for variant in gaVariants:
+            self.assertEqual(len(variant.calls), len(sampleIds))
+            for call in variant.calls:
+                self.assertIn(call.callSetName, sampleIds)
+
+    def _verifyVariantsCallSetIds(self, searchVariants, searchsampleIds):
+        """
+        SearchVariants = None, this will get all variants
+        If searchsampleIds = None, this will get all samples
+        else if = [of something], this will iterate through the list
+        """
+        gaCallSetVariants = []
         for referenceName in self._referenceNames:
             end = 2**30  # TODO This is arbitrary, and pysam can choke. FIX!
-            variants = list(self._gaObject.getVariants(
-                referenceName, 0, end, None, None))
-            allVariants += variants
+            gaVariants = list(self._gaObject.getVariants(
+                referenceName, 0, end, searchVariants, searchsampleIds))
+            self._verifyGaVariantsSample(gaVariants, searchsampleIds)
+            gaCallSetVariants += gaVariants
             localVariants = [
-                v for v in self._variants if v.CHROM == referenceName]
-            self.verifyVariantsEqual(variants, localVariants)
-        self.assertEqual(len(allVariants), len(self._variants))
+                v for v in self._variantRecords if (v.CHROM == referenceName)]
+            self._verifyVariantsEqual(gaVariants, localVariants)
+        if searchVariants is None:
+            self.assertEqual(len(gaCallSetVariants), len(self._variantRecords))
+
+    def testSearchAllVariants(self):
+        self._verifyVariantsCallSetIds(None, [])
+
+    def testSearchCallSetIdsSystematic(self):
+        for sampleIds in utils.powerset(self.vcfSamples, maxSets=10):
+            self._verifyVariantsCallSetIds(None, list(sampleIds))
+
+    def testVariantsValid(self):
+        end = 2**30  # TODO This is arbitrary, and pysam can choke. FIX!
```

Comment:
We should really just fix this at the source instead. Before we send a number to pysam, we should check that it's not going to choke on it. There's a wider issue of sanitisation for pysam inputs too, I guess. Another issue?